lib/build/nodeEngineFingerprint.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExtractFingerprint, FP, sha256 } from "@atomist/sdm-pack-fingerprints";
+import * as _ from "lodash";
+
+export const NodeEngineType = "nodeEngine";
+
+export interface Engine {
+  name: string;
+  version: string;
+}
+
+export declare type NodeEngineData = Engine[];
+/**
+ * Construct an engine fingerprint from the given engine and its version
+ * @param {string} name
+ * @param {string} version
+ * @return {FP}
+ */
+export function createEngineFingerprint(
+  name: string,
+  version: string,
+): FP<Engine> {
+  const data = { name, version };
+  return {
+    type: NodeEngineType,
+    name,
+    abbreviation: "engines",
+    version: "0.0.1",
+    data,
+    sha: sha256(JSON.stringify(data)),
+  };
+}
+
+export const nodeEngineFingerprint: ExtractFingerprint<
+  FP<NodeEngineData>
+> = async p => {
+  const pkg = await p.getFile("package.json");
+  if (!pkg) {
+    return undefined;
+  }
+
+  const jsonData = JSON.parse(await pkg.getContent());
```

Comment:
Probably best to wrap this in a `try`/`catch` block, since `JSON.parse` can throw errors.